lib/src/rules/no_duplicate_cases.dart

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.no_duplicate_cases;
+
+import 'dart:collection';
+import 'package:analyzer/context/declared_variables.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/src/dart/constant/evaluation.dart';
+import 'package:analyzer/src/dart/constant/value.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/resolver.dart';
+import 'package:linter/src/linter.dart';
+
+const desc = 'Do not use more the one case with same value';
+
+const details = r'''
+**DO NOT** use more the one case with same value. This can be
+of typo or changed value of constant.
+
+**GOOD:**
+
+```
+const int A = 1;
+switch (v) {
+  case A:
+  case 2:
+}
+```
+
+**BAD:**
+
+```
+const int A = 1;
+switch (v) {
+  case 1:
+  case 2:
+  case A:
+  case 2:
+}
+```
+''';
+
+class NoDuplicateCases extends LintRule {
+  NoDuplicateCases()
+      : super(
+            name: 'no_duplicate_cases',
```

Comment:
Consider naming it 'no_duplicate_case_values'. It seems a little less ambiguous to me.